app/validation.py

```diff
@@ -136,9 +138,13 @@ def get_validation_errors(validator_name, json_data,
             error_map[key] = _translate_json_schema_error(
                 key, error.validator, error.validator_value, error.message
             )
-        elif error.validator == 'required':
-            key = re.search(r'\'(.*)\'', error.message).group(1)
+        elif error.validator in ['required', 'dependencies']:
+            regex = r'u?\'(\w+)\' is a dependency of u?\'(\w+)\'' if error.validator == 'dependencies' else r'\'(.*)\''
```

Comment:
**tl:dr** Yes

Basically, you get two errors in both cases. 

Scenario 1:
```
# keys which depend on each other 
`["agileCoachPriceMin", "agileCoachPriceMax", "agileCoachLocations"]`

# submitted json
`{"agileCoachPriceMin": "100"}`

# unmodified error messages
errors = [{'message': "'agileCoachPriceMax' is a dependency of 'agileCoachPriceMin'"}, {'message': "'agileCoachLocations' is a dependency of 'agileCoachPriceMin'"}]
```

Scenario 2: 
```
# keys which depend on each other 
["agileCoachPriceMin", "agileCoachPriceMax", "agileCoachLocations"]

# submitted json
{"agileCoachPriceMin": "100", "agileCoachPriceMax": "101"}

# unmodified error messages
errors = [{'message': "'agileCoachLocations' is a dependency of 'agileCoachPriceMin'"}, {'message': "'agileCoachLocations' is a dependency of 'agileCoachPriceMax'"}]
```

Since we're just assigning errors to a dictionary and using the service key name, a second assignment to an already-existing key happens in some cases, but the practical result is the same. ie, in Scenario 2, `agileCoachLocations` gets overwritten with the same value, which seems fine.